### PR TITLE
Add limit on the log to 200 messages

### PR DESCRIPTION
**Pull request by rafaelje**
### Changes description

- Fix the error java.lang.OutOfMemoryError reported on Bugsnag
- Implement sqllite to store the log with a size of 200 rows on the table
- Improve the logs to show

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### Estimated time

|Assignee|:tomato:|
|:---|:---:|
|@rafaelje |21|

### References

Closes #N/A
Related #N/A
Depends on #N/A

### CODE REVIEW

**Comment by btry** [comment]
I suggest we switch to a sqlite table to not simply delete all log entries when tie limit is reached.

Using a sqlite table allows us to limit to a given count of entries and delete only the oldest extra entries, implementing a finite rotating log.

File: app/src/main/java/org/flyve/mdm/agent/data/database/dao/MDMLogDao.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @Insert
+    void insert(MDMLog... log);
+
+    @Update
+    void update(MDMLog... log);
+
+    @Query("SELECT * FROM log")
+    MDMLog[] loadAll();
+
+    @Query("DELETE FROM log")
```
[the remark this answers is not in the document]
Comment (rafaelje) [reply app/src/main/java/org/flyve/mdm/agent/data/database/dao/MDMLogDao.java]: This is the template by default and you are right I'm going to keep always the last messages

File: app/src/main/java/org/flyve/mdm/agent/data/database/entity/MDMLog.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    @PrimaryKey (autoGenerate = true)
+    @ColumnInfo (name = "id")
+    public int id;
+
+    @ColumnInfo (name = "description")
+    public String description;
+
+    @ColumnInfo (name = "type")
+    public String type;
+}
```
[the remark this answers is not in the document]
Comment (rafaelje) [reply app/src/main/java/org/flyve/mdm/agent/data/database/entity/MDMLog.java]: inside the log I got the date, but is easy keep outside too

**Comment by rafaelje** [comment]
Logs are working now with the sqllite, now improving the size and data
photo5150161345981818863

**Comment by rafaelje** [comment]
I'll test the agent tonight with this new login to close this PR tomorrow

**Pull request merged**